railgun/manager/run.go

```diff
@@ -88,14 +88,19 @@ func Run(ctx context.Context, c *config.Config) error {
 	}
 
 	// determine how to configure namespace watchers
-	if strings.Contains(c.WatchNamespace, ",") {
-		setupLog.Info("manager set up with multiple namespaces", "namespaces", c.WatchNamespace)
+	switch len(c.WatchNamespaces) {
+	case 0:
+		// watch all namespaces
+		controllerOpts.Namespace = corev1.NamespaceAll
+	case 1:
+		// watch one namespace
+		controllerOpts.Namespace = c.WatchNamespaces[0]
+	default:
 		// this mode does not set the Namespace option, so the manager will default to watching all namespaces
 		// MultiNamespacedCacheBuilder imposes a filter on top of that watch to retrieve scoped resources
 		// from the watched namespaces only.
-		controllerOpts.NewCache = cache.MultiNamespacedCacheBuilder(strings.Split(c.WatchNamespace, ","))
-	} else {
-		controllerOpts.Namespace = c.WatchNamespace
+		setupLog.Info("manager set up with multiple namespaces", "namespaces", c.WatchNamespace)
+		controllerOpts.NewCache = cache.MultiNamespacedCacheBuilder(c.WatchNamespaces)
```

Comment:
we need answer/be clear about the question:
do we watch whatever namespace user is configuring ?

Reply:
I'm not sure that I understand the question well but I'll try to answer as accurately as I can.

KIC 2.0 gives the user a possibility to specify:
- zero namespaces; in this case, KIC should watch all namespaces
- one namespace
- multiple namespaces

If the user specifies one namespaces, we watch only a single namespace for resources.
If the user specifies more than one namespace, due to the limitations of the k8s API we watch all the namespaces, and perform filtering client-side.

Reply:
possibly (most cases yes), we have a namespace black list - per user requirement, we dont allow to watch those namespaces coz they are user asset. (edited) 
10:33
What I understand atm, we don't have any limitation, but we need consider about it.